rancher2/structure_cluster_v2_rke_config_system_config.go

```diff
@@ -68,6 +68,82 @@ func flattenClusterV2RKEConfigSystemConfig(p []rkev1.RKESystemConfig) []interfac
 	return out
 }
 
+func flattenClusterV2RKEConfigMachineSelectorFiles(p []rkev1.RKEProvisioningFiles) []interface{} {
+	if p == nil {
+		return nil
+	}
+	out := make([]interface{}, len(p))
+	for i, in := range p {
+		obj := map[string]interface{}{}
+
+		if in.MachineLabelSelector != nil {
+			obj["machine_label_selector"] = flattenClusterV2RKEConfigSystemConfigLabelSelector(in.MachineLabelSelector)
+		}
+		if len(in.FileSources) > 0 {
+			obj["file_sources"] = flattenClusterV2RKEConfigFileSources(in.FileSources)
+		}
+		out[i] = obj
+	}
+
+	return out
+}
+
+func flattenClusterV2RKEConfigFileSources(p []rkev1.ProvisioningFileSource) []interface{} {
+	if p == nil {
+		return nil
+	}
+	out := make([]interface{}, len(p))
+	for i, in := range p {
+		obj := map[string]interface{}{}
+		obj["secret"] = flattenClusterV2RKEConfigK8sObjectFileSource(in.Secret)
```

Comment:
If the file source secret or configmap returns a nil object because the input is nil or subfields are, we don't want to set them I think? Terraform typically _removes_ nil fields from the tf config so they don't get passed to Rancher. I'd check to make sure this is the case

Reply:
Good points.
Checks for value are added.